rust/connlib/tunnel/src/device_channel/tun_android.rs

```diff
@@ -64,15 +56,10 @@ impl Tun {
         routes: HashSet<IpNetwork>,
         callbacks: &impl Callbacks,
     ) -> Result<()> {
-        let fd = callbacks
-            .on_update_routes(
-                routes.iter().copied().filter_map(ipv4).collect(),
-                routes.iter().copied().filter_map(ipv6).collect(),
-            )
-            .ok_or(Error::NoFd)?;
-
-        // SAFETY: we expect the callback to return a valid file descriptor
-        unsafe { self.replace_fd(fd)? };
+        let fd = callbacks.on_update_routes(
```

Comment:
```suggestion
        callbacks.on_update_routes(
```

Assignment of a unit-value.